src/tox/config.py

```diff
@@ -505,11 +505,32 @@ def setenv(testenv_config, value):
     )
 
     def basepython_default(testenv_config, value):
+        """Configure a sane interpreter for the environment.
+
+        If the environment contains a default factor, this will always be the
+        interpreter associated with that factor overriding anything manually
+        set.
+        """
+        for factor in testenv_config.factors:
+            if factor in tox.PYTHON.DEFAULT_FACTORS:
+                default = tox.PYTHON.DEFAULT_FACTORS[factor]
+
+                if value is None or testenv_config.config.ignore_basepython_conflict:
+                    return default
+
+                if str(value) != default:
+                    # TODO(stephenfin): Raise an exception here in tox 4.0
+                    print(
```

Comment:
let's use the warning system instead, https://docs.python.org/2/library/warnings.html; as the user can turn off that if they want

Reply:
Sounds good. I wasn't able to find any other examples of warnings in use so I assumed you didn't use them.